source/aws_imds_client.c

```diff
@@ -533,7 +538,10 @@ static void s_client_on_token_response(struct imds_user_data *user_data) {
         uint64_t expire_timestamp = aws_add_u64_saturating(
             current, aws_timestamp_convert(s_imds_token_ttl_secs, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_NANOS, NULL));
         s_update_token_safely(
-            user_data->client, cursor.len == 0 ? NULL : &user_data->imds_token, cursor.len != 0, expire_timestamp);
+            user_data->client,
+            cursor.len == 0 ? NULL : &user_data->imds_token,
```

Comment:
we checked that `user_data->current_result` is not empty and we have a 200, should we just error out if the cursor here is still empty?

Reply:
I have added an assert, as it should not be empty at this point.